src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/Functions/ProcessCompletionOfNextDataSetVersionImportFunctionTests.cs

```diff
@@ -117,6 +124,3168 @@ private static Mock<TaskOrchestrationContext> DefaultMockOrchestrationContext(Gu
+                                    s.SetOption(DataFixture.DefaultFilterOptionMeta())
+                                        .SetPublicId(SqidEncoder.Encode(3)))
+                                .Generate(1)))
+                .ForIndex(2, s =>
+                    s.SetPublicId(SqidEncoder.Encode(3))
+                        .SetOptionLinks(() =>
+                            DataFixture.DefaultFilterOptionMetaLink()
+                                .ForIndex(0, s =>
+                                    s.SetOption(DataFixture.DefaultFilterOptionMeta())
+                                        .SetPublicId(SqidEncoder.Encode(4)))
+                                .Generate(1)))
+                .GenerateList(3);
+
+            var (originalVersion, _) = await CreateDataSetInitialVersion(oldFilterMeta);
+
+            var newFilterMeta = DataFixture.DefaultFilterMeta()
+                .ForIndex(0, UnchangedFilterMetaSetter(oldFilterMeta[0])) // Filter and ALL options unchanged
+                .ForIndex(1, s =>
+                    s.SetPublicId(SqidEncoder.Encode(2)) // Filter changed
+                        .SetOptionLinks(() =>
+                            DataFixture.DefaultFilterOptionMetaLink()
+                                .ForIndex(0,
+                                    UnchangedFilterOptionMetaLinkSetter(oldFilterMeta[1]
+                                        .OptionLinks[0])) // Filter Option unchanged
+                                .ForIndex(1, s =>
+                                    s.SetOption(DataFixture.DefaultFilterOptionMeta())
+                                        .SetPublicId(SqidEncoder.Encode(5))) // Filter Option added
+                                .ForIndex(2, s =>
+                                    s.SetOption(DataFixture.DefaultFilterOptionMeta())
+                                        .SetPublicId(SqidEncoder.Encode(6))) // Filter Option added
+                                .Generate(3)))
+                .ForIndex(2, s =>
+                    s.SetPublicId(SqidEncoder.Encode(3)) // Filter changed
+                        .SetOptionLinks(() =>
+                            DataFixture.DefaultFilterOptionMetaLink()
+                                .ForIndex(0,
+                                    UnchangedFilterOptionMetaLinkSetter(oldFilterMeta[2]
+                                        .OptionLinks[0])) // Filter Option unchanged
+                                .ForIndex(1, s =>
+                                    s.SetOption(DataFixture.DefaultFilterOptionMeta())
+                                        .SetPublicId(SqidEncoder.Encode(7))) // Filter Option added
+                                .ForIndex(2, s =>
+                                    s.SetOption(DataFixture.DefaultFilterOptionMeta())
+                                        .SetPublicId(SqidEncoder.Encode(8))) // Filter Option added
+                                .Generate(3)))
```

Comment:
We want to test that the service code detects multiple option changes **within** a filter, and doesn't do something weird where it only detects 1 change, even if there are multiple. It may have a bug where it exits early etc...